### PR TITLE
Add OpenAI Finetunes connector

### PR DESCRIPTION
**Pull request by AathmanT**
# Description

Fixes https://github.com/wso2-enterprise/choreo/issues/20370

One line release note: 
Add OpenAI Finetunes connector

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Ballerina Version: 2201.4.1
* Operating System: Ubuntu 20.04.5 LTS
* Java SDK: openjdk 11.0.16 2022-07-19

# Checklist:

### Changes to OpenAPI definition

- [x] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [x] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [x] Added license header at the top of each .bal file. 
- [x] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [x] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [x] Added connector icon to the root of the connector. Icon needs to be a `png` of 200x200 px size with name `icon.png`
- [x] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [x] Compiled the connector successfully with the targeted ballerina version. 
- [x] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: openapi/openai.finetunes/Module.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+### Step 3: Invoke the connector operation
+1. Now, you can use the operations available within the connector. 
+
+ >**Note:** that they are in the form of remote operations.
+
+   Following is an example on fine tuning the GPT-3 ada model:
+
+   Fine tune ada model
```
Comment (NipunaRanasinghe) [inline openapi/openai.finetunes/Module.md:42]: Can we merge this 2 sentences. IMO the second sentence is redundant

Comment (AathmanT) [reply openapi/openai.finetunes/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/778/commits/ff5883d04f7c438dea2ea7ed609496ae44935e74

File: openapi/openai.finetunes/Module.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        auth: {
+            token: "sk-XXXXXXXXX"
+        }
+    });
+```
+
+### Step 3: Invoke the connector operation
+1. Now, you can use the operations available within the connector. 
+
+ >**Note:** that they are in the form of remote operations.
```
Comment (aneeshafedo) [inline openapi/openai.finetunes/Module.md:38]: What was the reason to mention like this ? The functions in the Client are resource functions not remote functions.

Comment (AathmanT) [reply openapi/openai.finetunes/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/778/commits/55047a5f4703accbd02297d226c748220a119199

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Review by aneeshafedo — APPROVED** [review]
LGTM

**Review by abeykoon — APPROVED** [review]

**Pull request merged**